src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/NormalizedPredicateReorderer.cs

```diff
@@ -26,6 +26,12 @@ public override Expression VisitSqlRoot(SqlRootExpression expression, object con
 
             List<TableExpression> reorderedExpressions = expression.TableExpressions.OrderByDescending(t =>
             {
+                // Sort _id (denormalized) expression to the front
```

Comment:
>// Sort _id (denormalized) expression to the front [](start = 16, length = 50)

Took me a while to figure out why it's working.

There is this code:
[https://github.com/microsoft/fhir-server/blob/1cb5628235aff99285e82a7ef193c153ad3c04ac/src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/DenormalizedPredicateRewriter.cs#L62](https://github.com/microsoft/fhir-server/blob/1cb5628235aff99285e82a7ef193c153ad3c04ac/src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/DenormalizedPredicateRewriter.cs#L62)
which does denormalization and put Table expression "All" on top of list of all expressions.

Which happened to be reordered here. That "All" expression doesn't have `SearchParameterQueryGenerator` so it get 0 from default statement in switch, and other table expression has ReferenceSearchParameterQueryGenerator which pushes it to the top of expression.

This weird condition `t.NormalizedPredicate == null && t.DenormalizedPredicate != null && t.Kind == TableExpressionKind.All`

comes directly from file which I linked above, but to be fair it looks like this is common pattern to create All table expression (at least in current code) so maybe it can be partially truncated.